File: redis/csharp/read/redis_read.cs
// -------------------------------------------------------------
//
//	redis_read.cs
//
//						Jun/16/2014
// -------------------------------------------------------------
using System;
using System.Collections.Generic;

using	Codeplex.Data;
// -------------------------------------------------------------
class redis_read
{

// -------------------------------------------------------------
static void Main (string[] args)
{
	Console.WriteLine ("*** 開始 ***");

	Redis rd = new Redis ("host_dbase",6379);

	List <String> keys = new List <String> ()
		{"t1851","t1852","t1853","t1854","t1855",
		"t1856","t1857","t1858","t1859"};

	foreach (String key in keys)
		{
		String json_str = null;

		try
			{
			json_str = rd.GetString (key);

			if (json_str != null)
				{
				var unit_bb = DynamicJson.Parse (json_str);

				Console.Write (key + "\t");
				Console.Write (unit_bb.name + "\t");
				Console.Write (unit_bb.population + "\t");
				Console.WriteLine (unit_bb.date_mod + "\t");
				}
			}
		catch	(Exception exp)
			{
		Console.Error.WriteLine ("*** error *** " + exp);
			}

		}

	Console.WriteLine ("*** 終了 ***");
}

// -------------------------------------------------------------
}
// -------------------------------------------------------------

File: redis/typescript/read/redis_read.ts
// ---------------------------------------------------------------
//	redis_read.ts
//
//					Feb/09/2016
//
// ---------------------------------------------------------------
declare function require(x: string): any;
declare var process:any;

var redis = require ("redis")

console.log ("*** 開始 ***")

var client = redis.createClient(6379,'host_dbase')

client.on ("error", function (err)
{
        console.log ("Redis connection error to "
		+ client.host + ":" + client.port + " - " + err)
})

var keys=["t1851","t1852","t1853",
	"t1854","t1855","t1856",
	"t1857","t1858","t1859"]
for (var it in keys)
	{
	read_single_proc (client,keys[it])
	}

client.quit()
console.log ("*** 終了 ***")

// ---------------------------------------------------------------
function read_single_proc (client,key)
{
	client.get (key, function (err, reply)
	{
	if (err)
		{
		console.log("Get error: " + err)
        	}
	 else if (reply != null)
		{
		var json_str = reply
		var data = JSON.parse (json_str)

		var out_str: string = key + "\t"
		out_str  += data.name + "\t"
		out_str += data.population + "\t"
		out_str += data.date_mod
		console.log (out_str)
		}
	})
}

// ---------------------------------------------------------------
